app/screens/channel_info/channel_info.js

```diff
@@ -402,6 +436,20 @@ export default class ChannelInfo extends PureComponent {
         return isDirectMessage || isGroupMessage;
     };
 
+    renderUnarchiveChannel = () => {
```

Comment:
the function is to determine if the option should be rendered or not, maybe it should be 
```suggestion
    shouldRenderRestoreChannel = () => {
```